mm2src/coins/eth/nft_swap_v2/mod.rs

```diff
@@ -37,11 +37,15 @@ impl EthCoin {
                 let htlc_data = try_tx_s!(self.prepare_htlc_data(&args));
 
                 let data = try_tx_s!(self.prepare_nft_maker_payment_v2_data(&args, htlc_data).await);
+                let gas_limit = match args.nft_swap_info.contract_type {
```

Comment:
Could we refactor getting gas_limit (wherever it is obtained) to do it in a way like that:
```
   let gas_limit = self.get_gas_limit_for_swap(&args)?;
```
This would make swap methods more compact.
I guess, for that some trait may be implemented for args that extracts the appropriate gas limit from EthCoin::gas_limit_v2

Reply:
Provided gas limit traits https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/b3ed7bacd700357a10c42418d43c4022986ecb29